### PR TITLE
Conveyor : Protection to prevent getting files from GCS : Closes #3510

### LINKED ISSUE #3510: Protection to prevent getting files from GCS
Motivation
----------
A protection needs to be added so that only privileged users can get data from GCS

### PR DESCRIPTION
**Pull request by cserf**
Protection to prevent getting files from GCS : Closes #3510

### CODE REVIEW

**Review by bziemons — APPROVED** [review]
lgtm

**Comment by cserf** [comment]
Up

**Comment by cserf** [comment]
ping @bari12

File: lib/rucio/core/transfer.py [hunk 1]
```diff
@@ -664,6 +665,7 @@ def get_transfer_requests_and_source_replicas(total_workers=0, worker_number=0,
     bring_online_local = bring_online
     transfers, rses_info, protocols, rse_attrs, reqs_no_source, reqs_only_tape_source, reqs_scheme_mismatch = {}, {}, {}, {}, [], [], []
     multi_hop_dict = {}
+    admin_accounts = [acc['account'].external for acc in list_accounts(filter={'admin': True}, session=session)]
```
Comment (bari12) [inline lib/rucio/core/transfer.py:668]: Let's cache this in memcache as well.

**Review by bari12 — COMMENTED** [review]
Apologies for the delay @cserf 
Still not entirely happy about all these magic variables (restricted_read/write etc. but I also don't have a better idea)
Just one quick comment, we should cache the `list_accounts` result as well in memache. The `parse_expression` is already cached by the method natively. But we will call the list_accounts a lot too then.

**Comment by bari12** [comment]
Needs rebase + please still add the caching.

**Comment by rcarpa** [comment]
This one will require quite a rebase. I'd say: a complete rewrite

**Comment by bari12** [comment]
@cserf should we close this for now? It will really require some rewrite with the changes @rcarpa made in the meanwhile

**Pull request closed without merging**